components/script/body.rs

```diff
@@ -127,10 +130,17 @@ impl TransmitBodyConnectHandler {
     /// in favor of making `re_extract` actually re-extract a stream from the source.
     /// See #26686
     fn transmit_source(&self) {
```

Comment:
I think we can make this take a `&mut self`, and then we would not need the cell.

Reply:
Okay!

actually I personally prefer not to use `mut` because it can help us make sure the _user_ who call this function won't mutate anything to it 👀 but I'm fine to change it! let me update!